### ISSUE TITLE
Access token never expiring

### ISSUE THREAD

**Issue by carlosmojopicon** [thephpleague/oauth2-server#559]
I'm successfully doing client grants and getting access tokens back. Tokens are generated with 10 seconds TTL for test purposes.

The problem comes when I use generated tokens to do authenticated calls. Everything works fine, but tokens are never expiring. It responds with access_denied if I pass a fake token which is ok, but if I use a valid one it always pass the validation even if it's expired. Am I missing something?

**Comment by alexbilbie** [comment]
Which version are you using? There was an issue resolved with non-expiring
tokens in RC1 which was resolved in RC2.

Can you provide any example code to demonstrate?

**Comment by carlosmojopicon** [comment]
I was using RC1. Updated to 5.0.2 but new problems came.

After solving some namespace changes, I ended up with a Fatal error:  Call to a member function setClient() on null in /vendor/league/oauth2-server/src/Grant/AbstractGrant.php on line 304 when invoking respondToAccessTokenRequest for a client grant.

It seems to be due to I didn't implement new getNewToken method from AccessTokenRepositoryInterface. The question is how to do that? Do I need now to generate access tokens by myself? Do I need to implement AccessTokenEntityInterface and all TokenInterface methods to be able to generate and return a AccessTokenEntity within getNewToken method? I'm so confused now with this latest release.
